### PR TITLE
Image and Color classes implementations

### PR DESCRIPTION
**Pull request by tomczylok**
#26

### CODE REVIEW

**Comment by dkrawiec** [comment]
What about flyweight design pattern approach? What about builder?

File: imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package pl.edu.uj.kimage;
+
+public final class Image {
+    private final Color[] data;
+    private final int width;
+    private final int height;
+
+    public Image() {
```
Comment (dkrawiec) [inline imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java]: What is the purpose of this constructor?

Comment (tomczylok) [inline imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java]: I have added in only to compatibility with other code (there was already use of Image with empty constructor so far).

**Comment by tomczylok** [comment]
I suggest implementing flyweight pattern in seperate task, because we need this class Image as soon as possible and I reckon that current form is sufficient.

**Comment by dkrawiec** [comment]
Sure. Let's do it together with builder

EDIT: Ok, I decided that stories should be small and independent. It should be covered by  #40 #41

File: imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java (hunk 1 above)
Comment (dkrawiec) [inline imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java]: Yes, in tests ;) Please adjust it and remove the constructor.

Comment (tomczylok) [inline imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java]: Ok, I also noticed that we should not use null anywhere, so my mistake. Thanks.

**Pull request merged**